### PR TITLE
Change font to a font stored in the jar

### PR DESCRIPTION
**Pull request by UnsolvedCypher**
Changes the font from Courier to Adobe Sourcecode Pro, which is stored in the jar (so that it looks the same across all systems no matter what font is installed).

### CODE REVIEW

**Comment by jfly** [comment]
Awesome! This is definitely doing something on my laptop:

With your changes:

2015-04-21_2560x1413_kaladin

Without your changes:

2015-04-21_2560x1413_kaladin

I see `./tmt make run -p wca` fail with a NPE. This gets sent to the browser when trying to generate a zip or pdf:

```
java.lang.NullPointerException
       at net.gnehzr.tnoodle.server.webscrambles.ScrambleRequest.splitScrambleToLineChunks(ScrambleRequest.java:670)
       at net.gnehzr.tnoodle.server.webscrambles.ScrambleRequest.createTable(ScrambleRequest.java:831)
       at net.gnehzr.tnoodle.server.webscrambles.ScrambleRequest.addScrambles(ScrambleRequest.java:572)
       at net.gnehzr.tnoodle.server.webscrambles.ScrambleRequest.createPdf(ScrambleRequest.java:243)
       at net.gnehzr.tnoodle.server.webscrambles.ScrambleRequest.requestsToPdf(ScrambleRequest.java:1096)
       at net.gnehzr.tnoodle.server.webscrambles.ScrambleViewHandler.wrappedService(ScrambleViewHandler.java:236)
       at net.gnehzr.tnoodle.server.SafeHttpServlet.service(SafeHttpServlet.java:41)
       at javax.servlet.http.HttpServlet.service(HttpServlet.java:45)
       at winstone.ServletConfiguration.execute(ServletConfiguration.java:249)
       at winstone.RequestDispatcher.forward(RequestDispatcher.java:335)
       at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:378)
       at net.gnehzr.tnoodle.server.HtmlInjectFilter.doFilter(HtmlInjectFilter.java:40)
       at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)
       at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)
       at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176)
       at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145)
       at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92)
       at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:389)
       at winstone.FilterConfiguration.execute(FilterConfiguration.java:195)
       at winstone.RequestDispatcher.doFilter(RequestDispatcher.java:368)
       at winstone.RequestDispatcher.forward(RequestDispatcher.java:333)
       at winstone.RequestHandlerThread.processRequest(RequestHandlerThread.java:244)
       at winstone.RequestHandlerThread.run(RequestHandlerThread.java:150)
       at java.lang.Thread.run(Thread.java:662)
```

**Comment by UnsolvedCypher** [comment]
Here is what it looks like on my computer:

before:
[image: Immagine in linea 1]
after:
[image: Immagine in linea 2]

your computer is most likely defaulting to the default font because it
can't find Courier, so your font is much closer to Sourcecode Pro.

2015-04-21 15:42 GMT-04:00 Jeremy Fleischman <notifications@github.com>:

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 1]
```diff
@@ -797,19 +797,20 @@ private static TableAndHighlighting createTable(PdfWriter docWriter, Document do
         Font scrambleFont = null;
 
         try {
-            BaseFont courier = BaseFont.createFont(BaseFont.COURIER, BaseFont.CP1252, BaseFont.EMBEDDED);
+            // load the font from the resource folder. This loads the regular version, but can be changed to light, bold, etc if people prefer that
+            BaseFont sourcecode = BaseFont.createFont(Utils.getResourceDirectory().getPath() + "/fonts/SourceCodePro-Regular.ttf", BaseFont.CP1252, BaseFont.EMBEDDED);
```
Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: I know we previously called this variable `courier`, but I think we should call this `sourceCodeFont`. `courier` is obviously a font to me, whereas `sourcecode` has all sorts of other meanings =)

File: webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java [hunk 2]
```diff
@@ -797,19 +797,20 @@ private static TableAndHighlighting createTable(PdfWriter docWriter, Document do
         Font scrambleFont = null;
 
         try {
-            BaseFont courier = BaseFont.createFont(BaseFont.COURIER, BaseFont.CP1252, BaseFont.EMBEDDED);
+            // load the font from the resource folder. This loads the regular version, but can be changed to light, bold, etc if people prefer that
```
Comment (jfly) [inline webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java]: nit: For consistency's sake, lets do full sentences (capital letter at the start, and periods at the end).

**Comment by jfly** [comment]
Hey, any news here? It would be awesome to push this one over the finish line.

**Comment by jfly** [comment]
I'm going to close this PR, feel free to reopen if you're motivated to fix this.

**Pull request closed without merging**